bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java

```diff
@@ -689,20 +705,22 @@ public void initialize() {
                     "@text/offline.conf-error-no-ip-address");
         } else {
             if (hueBridge == null) {
-                hueBridge = new HueBridge(ip, hueBridgeConfig.getPort(), hueBridgeConfig.getProtocol(), scheduler);
+                if (HueBridgeConfig.HTTPS.equals(hueBridgeConfig.getProtocol())) {
+                    // register trustmanager service
+                    HueTlsTrustManagerProvider tlsTrustManagerProvider = new HueTlsTrustManagerProvider(
+                            ip + ":" + hueBridgeConfig.getPort(), hueBridgeConfig.useSelfSignedCertificate());
+                    serviceRegistration = FrameworkUtil.getBundle(getClass()).getBundleContext()
+                            .registerService(TlsTrustManagerProvider.class.getName(), tlsTrustManagerProvider, null);
```

Comment:
This is a really ugly implementation and we do not want it. But unfortunately I did not yet found a better solution.

Implementing the `HueTlsTrustManagerProvider` to be a `ThingHandlerService` would be my favorite on but when doing so there will be a race condition between activation of the `HueTlsTrustManagerProvider` and initialization of the `HueBridgeHandler`.

Registering the ``HueTlsTrustManagerProvider` in the `HueThingHandlerFactory` does not work because I need access to the configuration of the `Thing` to retrieve IP address, port and protocol.

Reply:
Is there a way for HueTlsTrustManagerProvider to return a status in the case that getting the external cert fails (i.e. the server is momentarily unreachable)? It seems like the current code has no provision to retry the cert retrieval process since it attempted only once during initialize().